sycl/unittests/pi/piPlatformTest.cpp

```diff
@@ -0,0 +1,104 @@
+//==---- piPlatformTest.hpp --- PI unit tests
+//-------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "CL/sycl/detail/pi.hpp"
+#include <CL/sycl.hpp>
+#include <gtest/gtest.h>
+#include <memory>
+
+using namespace cl::sycl;
+
+namespace {
+class piPlatformTest : public ::testing::Test {
+protected:
+  piPlatformTest() { detail::pi::piInitialize(); }
+
+  ~piPlatformTest() = default;
+};
+
+TEST_F(piPlatformTest, piPlatformGetInfo) {}
+
+TEST_F(piPlatformTest, piPlatformsGet) {
+  pi_uint32 platformCount = 0;
+
+  ASSERT_EQ(PI_CALL_RESULT(RT::piPlatformsGet(0, 0, &platformCount)),
+            PI_SUCCESS)
+      << "piPlatformsGet failed";
+
+  ASSERT_TRUE(platformCount > 0) << "piPlatformsGet found 0 platforms.\n";
+
+  std::vector<pi_platform> platforms(platformCount);
+
+  ASSERT_EQ(PI_CALL_RESULT(
+                RT::piPlatformsGet(platformCount, platforms.data(), nullptr)),
+            PI_SUCCESS)
+      << "piPlatformsGet failed with nullptr for return size.\n";
+}
+
+constexpr size_t out_string_size =
+    8192u; // Using values from OpenCL CTS clGetPlatforms test
+
+// this should be part of the PI API
```

Comment:
Maybe you should put it somewhere around [here](https://github.com/intel/llvm/blob/sycl/sycl/source/detail/pi.cpp) then?

Reply:
Yes I can do. I was thinking [pi.hpp](https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/detail/pi.hpp) might be a good idea so that it is easily accessible.

Reply:
I have moved this function to pi.hpp!